### PR TITLE
Feature/MoEngage integration

### PR DESCRIPTION
**Pull request by ariefwijaya**
## Purpose
- Added capabilities to integrate with MoEngage such as Amplitude
- ~~Fix linting issues and remove codesmells~~ , [Update, gonna create new PR for this purpose]
- closes #79 

## Types of changes
- [X] New feature (non-breaking change which adds functionality)
- [X] Documentation Update

## How to Test
- just add `moengageIntegrationEnabled:true` in the dart code when trying to integrate MoEngage with segment

## Environment
- [X] Development
- [X] Android
- [X] iOS

<details><summary> Attach your `flutter doctor -v` inside code block: </summary>
<p>

```
✓] Flutter (Channel stable, 2.5.2, on macOS 11.6 20G165 darwin-x64, locale en-ID)
    • Flutter version 2.5.2 at /Users/ittcel/fvm/versions/stable
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 3595343e20 (8 days ago), 2021-09-30 12:58:18 -0700
    • Engine revision 6ac856380f
    • Dart version 2.14.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/ittcel/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 13.0, Build version 13A233
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2020.3.4)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.60.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.27.0

[✓] Connected device (2 available)
    • Android SDK built for x86 64 (mobile) • emulator-5554 • android-x64    • Android 9 (API 28) (emulator)
    • Chrome (web)                          • chrome        • web-javascript • Google Chrome 94.0.4606.71

• No issues found!
```

</p>
</details>

## Expected Results
- Integrated with MoEngage
- All tracked data will be send to MoEngage from segment

### CODE REVIEW

**Comment by danielgomezrico** [comment]
@ariefwijaya please do a rebase after you apply your changes, in this PR https://github.com/claimsforce-gmbh/flutter-segment/pull/95 the CI will be good to see if this does not break anything :)

**Comment by MaiKaY** [comment]
> @ariefwijaya please do a rebase after you apply your changes, in this PR #95 the CI will be good to see if this does not break anything :)

@ariefwijaya just a friendly reminder :-D

**Comment by ariefwijaya** [comment]
> > @ariefwijaya please do a rebase after you apply your changes, in this PR #95 the CI will be good to see if this does not break anything :)
> 
> @ariefwijaya just a friendly reminder :-D

Sorry for the late reply, Please recheck 
@MaiKaY @danielgomezrico

**Review by danielgomezrico — APPROVED** [review]
It looks good to me, check the linter because there are some errors that must be fixed before merging it :)

**Comment by ariefwijaya** [comment]
> Hi, Let me fix the CI before we merge this, so we can be sure that everything compiles.
> 
> @ariefwijaya I have some suggestions in the meantime:
> 
> * Can you move the linter fixes into another PR? So this one remains only with the new integration?
> * I see issues with the linter which can be fixed in the while, but that will become clear after you move all of that code to another PR

@danielgomezrico I removed the linter fixes to another PR based on your request

File: analysis_options.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ linter:
 analyzer:
   strong-mode:
     # TODO: enable
-    #implicit-casts: false
```
Comment (danielgomezrico) [inline analysis_options.yaml:14]: Can we let this lint commented? it will cause a fail in the current code lint

**Review by danielgomezrico — APPROVED** [review]
Beside removing the lint setup it looks great to me, can you please fix that before merge?

**Comment by danielgomezrico** [comment]
@ariefwijaya I believe that the CI got broken because of 
image

but I opened a PR to fix that https://github.com/claimsforce-gmbh/flutter-segment/pull/108, I'm sorry for been so picky about these things, but it helps in the future to have a better code base :)

**Comment by MaiKaY** [comment]
This project is not longer maintained from this repository. In case you want to merge your PR please create a new PR [here](https://github.com/la-haus/flutter-library-segment/pulls).

**Pull request closed without merging**